aws/convert_types.go

```diff
@@ -846,7 +846,7 @@ func TimeValue(v *time.Time) time.Time {
 // representing seconds since Epoch or time.Time{} if the pointer is nil.
 func SecondsTimeValue(v *int64) time.Time {
 	if v != nil {
-		return time.Unix((*v / 1000), 0)
+		return time.Unix(*v, 0)
 	}
 	return time.Time{}
 }
```

Comment:
I took a look back at the the original pull request that proposed the implementation of these functions originally. That can be found here [#1433](https://github.com/aws/aws-sdk-go/pull/1433). The documented intention of these functions was to take millisecond `int64` time values (like those returned by CloudWatch Logs for example) and provide convenience functions for converting to a time.Time value. The hence the names imply the level of granularity in which the `time.time` value will be constructed with.  With that said I think it would be better to document that this is the intended behavior and make it clear in the function docstring.

Reply:
I'm very surprised and frustrated that this is the intended behavior.

Some AWS APIs return time values as seconds since epoch (see the [SQS `GetQueueAttributes` API](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_GetQueueAttributes.html), especially the `CreatedTimestamp` and `LastModifiedTimestamp` attributes).

[`MillisecondsTimeValue`](https://pkg.go.dev/github.com/aws/aws-sdk-go/aws?tab=doc#MillisecondsTimeValue) exists, as does [`(time.Time).Truncate`](https://pkg.go.dev/time?tab=doc#Time.Truncate) (in the cases where you really want to truncate a `time.Time` value).

Given all that, I would expect a similar helper for `*int64` values that contain seconds since epoch. It's disappointing that `SecondsTimeValue` isn't that.

Reply:
Additionally, [this conversation on that PR](https://github.com/aws/aws-sdk-go/pull/1433#discussion_r130436128) reads to me like the intention was to have a function that take epoch milliseconds, and another that takes epoch seconds.